Emby.Server.Implementations/Devices/DeviceManager.cs

```diff
@@ -194,172 +170,6 @@ private string GetDevicePath(string id)
-                }
-
-                AddCameraUpload(deviceId, file);
-            }
-            finally
-            {
-                _libraryMonitor.ReportFileSystemChangeComplete(path, true);
-            }
-
-            if (CameraImageUploaded != null)
-            {
-                CameraImageUploaded?.Invoke(this, new GenericEventArgs<CameraImageUploadInfo>
-                {
-                    Argument = new CameraImageUploadInfo
-                    {
-                        Device = device,
-                        FileInfo = file
-                    }
-                });
-            }
-        }
-
-        private void AddCameraUpload(string deviceId, LocalFileInfo file)
-        {
-            var path = Path.Combine(GetDevicePath(deviceId), "camerauploads.json");
-            Directory.CreateDirectory(Path.GetDirectoryName(path));
-
-            lock (_cameraUploadSyncLock)
-            {
-                ContentUploadHistory history;
-
-                try
-                {
-                    history = _json.DeserializeFromFile<ContentUploadHistory>(path);
-                }
-                catch (IOException)
-                {
-                    history = new ContentUploadHistory
-                    {
-                        DeviceId = deviceId
-                    };
-                }
-
-                history.DeviceId = deviceId;
-
-                var list = history.FilesUploaded.ToList();
-                list.Add(file);
-                history.FilesUploaded = list.ToArray();
-
-                _json.SerializeToFile(history, path);
-            }
-        }
-
-        internal Task EnsureLibraryFolder(string path, string name)
-        {
-            var existingFolders = _libraryManager
-                .RootFolder
-                .Children
-                .OfType<Folder>()
-                .Where(i => _fileSystem.AreEqual(path, i.Path) || _fileSystem.ContainsSubPath(i.Path, path))
-                .ToList();
-
-            if (existingFolders.Count > 0)
-            {
-                return Task.CompletedTask;
-            }
-
-            Directory.CreateDirectory(path);
-
-            var libraryOptions = new LibraryOptions
-            {
-                PathInfos = new[] { new MediaPathInfo { Path = path } },
-                EnablePhotos = true,
-                EnableRealtimeMonitor = false,
-                SaveLocalMetadata = true
-            };
-
-            if (string.IsNullOrWhiteSpace(name))
-            {
-                name = _localizationManager.GetLocalizedString("HeaderCameraUploads");
-            }
-
-            return _libraryManager.AddVirtualFolder(name, CollectionType.HomeVideos, libraryOptions, true);
-        }
-
-        private Tuple<string, string, string> GetUploadPath(DeviceInfo device)
-        {
-            var config = _config.GetUploadOptions();
-            var path = config.CameraUploadPath;
```

Comment:
Looks like `MediaBrowser.Model.Devices.DevicesOptions` can be remove now as well

Reply:
DeviceOptions still has usages throught the code. If it can be removed, I think it should be done in its own PR

Reply:
I was referring to Device**s**Options, with an s. It's not used any more.